Gruntfile.js

```diff
@@ -157,19 +231,87 @@ module.exports = function(grunt) {
             'ui-common': {
                 src: 'build/test-coverage/lcov/**/*.info',
             }
-        }
+        },
+        // file concatenation
+        concat: {
+            options: {
+                banner: cssFileHeader,
+            },
+            // css for the "page" template
+            pageCss: {
+                src: pageCssFiles,
+                dest: kbaseStaticDir + 'css/page_concat.css',
+                nonull: true,
+            },
+            // a single file containing all the KBase css!
+            allCss: {
+                src: allCssFiles,
+                dest: kbaseStaticDir + 'css/all_concat.css',
+                nonull: true,
+            }
+        },
 
+        // check the source css files are not completely horrendous
+        stylelint: {
+            options: stylelintOptions,
+            kbaseCss: {
+                src: staticFilesNoConcat,
+            },
+        },
+
+        // Run CSS / SCSS-related tasks
+        // these all modify files in place
+        postcss: {
+            // clean up the files in kbase-extension/static/kbase/(css|custom)/*.css
+            kbaseCss: {
+                options: cleanExistingCssOptions,
+                src: staticFilesNoConcat,
+            },
+            // autoprefix and minify the concatenated css files
+            concat: {
+                options: {
+                    processors: [
+                        // add vendor prefixes
+                        require('autoprefixer')(),
+                        // add minification later
+                        // require('cssnano')([
+                        //     "default",
+                        //     {
+                        //         "normalizeWhitespace": {
+                        //             "exclude": true
+                        //         },
+                        //     }
+                        // ]),
+                    ],
+                },
+                src: [
+                    'kbase-extension/static/kbase/css/*_concat.css',
+                ],
+            }
+        },
+
+        // watch css files for any changes
+        // when they change, regenerate the concat'd css files
+        watch: {
+            files: staticFilesNoConcat,
+            tasks: [
+                'concat',
+                'postcss:concat'
+            ],
+        },
     });
 
-    grunt.registerTask('minify', [
-        'requirejs',
-        'uglify',
-        'regex-replace'
+    grunt.registerTask('build_css', [
+        'postcss:kbaseCss',
+        // run stylelint --fix
+        'stylelint:kbaseCss',
+        'concat',
+        'postcss:concat'
     ]);
 
-    grunt.registerTask('build', [
+    grunt.registerTask('minify', [
         'requirejs',
-        'filerev',
+        'uglify',
```

Comment:
`filerev` is deprecated, so I removed it from the `package.json` and removed any tasks referring to it.